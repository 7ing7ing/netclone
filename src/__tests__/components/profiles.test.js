import React from "react";
import { render } from "@testing-library/react";
import { Profiles } from "../../components";

describe("<Profiles />", () => {
  it("Renders the <Profiles/> with populated data", () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture
              src="/images/ting.png"
              data-testid="profile-picture"
            />
            <Profiles.Name>Ting-Ting Chen</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );
    expect(getByText("Who's watching?"));
    expect(getByTestId("profile-picture")).toBeTruthy();
    expect(getByText("Ting-Ting Chen")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders the <Profiles/> with populated data but misc profile image", () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture
              src="/images/ting.png"
              data-testid="profile-picture-misc"
            />
            <Profiles.Name>Ting-Ting Chen</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );
    expect(getByText("Who's watching?"));
    expect(getByTestId("profile-picture-misc")).toBeTruthy();
    expect(getByText("Ting-Ting Chen")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
