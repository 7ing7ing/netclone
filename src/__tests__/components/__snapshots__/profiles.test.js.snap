// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> Renders the <Profiles/> with populated data 1`] = `
<div
  class="profiles__Container-sc-bxtkkv-0 khxgvE"
>
  <h1
    class="profiles__Title-sc-bxtkkv-1 dhTqjW"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-sc-bxtkkv-2 kcDWLv"
  >
    <li
      class="profiles__Item-sc-bxtkkv-5 jHZFyh"
    >
      <img
        class="profiles__Picture-sc-bxtkkv-4 cyFckY"
        data-testid="profile-picture"
        src="/images/users//images/ting.png.png"
      />
      <p
        class="profiles__Name-sc-bxtkkv-3 hByAmj"
      >
        Ting-Ting Chen
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> Renders the <Profiles/> with populated data but misc profile image 1`] = `
<div
  class="profiles__Container-sc-bxtkkv-0 khxgvE"
>
  <h1
    class="profiles__Title-sc-bxtkkv-1 dhTqjW"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-sc-bxtkkv-2 kcDWLv"
  >
    <li
      class="profiles__Item-sc-bxtkkv-5 jHZFyh"
    >
      <img
        class="profiles__Picture-sc-bxtkkv-4 cyFckY"
        data-testid="profile-picture-misc"
        src="/images/users//images/ting.png.png"
      />
      <p
        class="profiles__Name-sc-bxtkkv-3 hByAmj"
      >
        Ting-Ting Chen
      </p>
    </li>
  </ul>
</div>
`;
